test/test_profiler.py

```diff
@@ -1192,5 +1140,109 @@ def find_ones(nodes):
             torch._C._autograd._ExtraFields_Allocation)
 
 
+class TestExperimentalUtils(TestCase):
+
+    @staticmethod
+    def generate_mock_profile():
```

Comment:
This is an absolutely beautiful method. You should include a unit test that utilizes it.

Reply:
I see you use it in later PRs to test the new functionality, but you should also use it to validate the existing functionality.

Reply:
Okay understood

Reply:
I added a temporary test (with a list of expected digits) because I'm missing some information in qd list to print GPU/CPU/cudaLaunch in this commit. This should be enough to verify correctness tho.